src/parser/parsed_expression_visitor.cpp

```diff
@@ -0,0 +1,20 @@
+#include "parser/parsed_expression_visitor.h"
+
+namespace kuzu {
+namespace parser {
+
+std::unique_ptr<ParsedExpression> ParsedExpressionVisitor::replaceMacroParameters(
```

Comment:
This is not the exact way to write a visitor.

I think you want ParsedExpressionVisitor to only define `getChildren()` interface because some expression, e.g. `CASE` children are stored separately inside of children directly. 

And you want an other class like ParsedExpressionParameterReplacor to implement such logic.